code/modules/reagents/chemistry/holder/holder.dm

```diff
@@ -462,20 +462,19 @@
 
 ///Remove a reagent datum with the type provided from this container. True if one is removed, false otherwise.
 /datum/reagents/proc/del_reagent(type_to_remove)
-	var/datum/reagent/reagent_to_remove = locate(type_to_remove) in reagent_list
-	if(!reagent_to_remove)
-		return FALSE
-	SEND_SIGNAL(src, COMSIG_REAGENT_DELETING, type_to_remove)
-	var/atom/holder_atom = get_holder()
-	if(isliving(holder_atom))
-		var/mob/living/L = holder_atom
-		reagent_to_remove.on_mob_delete(L, L.get_reagent_tags())
-	reagent_list -= reagent_to_remove
-	qdel(reagent_to_remove)
-	update_total()
-	holder_atom?.on_reagent_change(DEL_REAGENT)
-	return TRUE
-
+	for(var/datum/reagent/reagent_to_remove AS in reagent_list)
```

Comment:
You can use get_reagent for this. Same functionality, but with bonus proc overhead.